File: server/models/WeeklyReport.js
const mongoose = require('mongoose');

const WeeklyReportSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true
    },
    weekStartDate: {
      type: Date,
      required: true
    },
    weekEndDate: {
      type: Date,
      required: true
    },
    summary: {
      type: String,
      default: ''
    },
    highlights: {
      type: [String],
      default: []
    },
    improvements: {
      type: [String],
      default: []
    },
    recommendations: {
      type: [String],
      default: []
    },
    productivityScore: {
      type: Number,
      default: 0
    }
  },
  { timestamps: true }
);

WeeklyReportSchema.index({ userId: 1, weekStartDate: -1 });

module.exports = mongoose.model('WeeklyReport', WeeklyReportSchema);

File: flutter_client/lib/features/reports/models/weekly_report.dart
class WeeklyReport {
  final String id;
  final DateTime weekStartDate;
  final DateTime weekEndDate;
  final String summary;
  final List<String> highlights;
  final List<String> improvements;
  final List<String> recommendations;
  final num productivityScore;

  WeeklyReport({
    required this.id,
    required this.weekStartDate,
    required this.weekEndDate,
    this.summary = '',
    this.highlights = const [],
    this.improvements = const [],
    this.recommendations = const [],
    this.productivityScore = 0,
  });

  factory WeeklyReport.fromJson(Map<String, dynamic> json) {
    return WeeklyReport(
      id: json['_id'] as String,
      weekStartDate: DateTime.parse(json['weekStartDate'] as String),
      weekEndDate: DateTime.parse(json['weekEndDate'] as String),
      summary: json['summary'] as String? ?? '',
      highlights: (json['highlights'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      improvements: (json['improvements'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      recommendations: (json['recommendations'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      productivityScore: json['productivityScore'] as num? ?? 0,
    );
  }
}
